### PR TITLE
New logging library 

### PR DESCRIPTION
**Pull request by rish9511**
Introduce new Logging library: https://github.com/gabime/spdlog

### CODE REVIEW

File: 3rdparty/ErrorHandling/MavenException.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "MavenException.h"
+#include <elmavenlogger.h>
```
Comment (sahil21) [inline 3rdparty/ErrorHandling/MavenException.cpp]: @rish9511 Why not move this include statements in the header file?

File: 3rdparty/Logger/elmavenlogger.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    catch(const spdlog::spdlog_ex& exp) {
+        
+        std::cout << "failed initializing the logger: " << exp.what() ;
+    }
+    
+
+}
+
+void ElMavenLogger::init()
```
Comment (sahil21) [inline 3rdparty/Logger/elmavenlogger.cpp]: @rish9511 What is this for?

File: 3rdparty/ErrorHandling/MavenException.cpp [hunk 3]
```diff
@@ -9,7 +10,7 @@ MavenException::MavenException(ErrorCodes::Errors errCd, const std::string& deta
     message += details;
     message += "\n";
 
-    LOGD << message;
+    ElMavenLogger::getInstance()->logErr(message, ElMavenLogger::info);
```
Comment (sahil21) [inline 3rdparty/ErrorHandling/MavenException.cpp]: @rish9511 Why as `ElMavenLogger::info` and not something like critical or fatal?

File: 3rdparty/ErrorHandling/MavenException.cpp (hunk 1 above)
Comment (rish9511) [reply 3rdparty/ErrorHandling/MavenException.cpp]: @sahil21  we should avoid including files in the header files as much as we can. 
Compilation time increases if the files are included in header files

File: 3rdparty/Logger/elmavenlogger.cpp (hunk 2 above)
Comment (rish9511) [reply 3rdparty/Logger/elmavenlogger.cpp]: @sahil21 
just for better readability. It initializes our logger class

File: 3rdparty/ErrorHandling/MavenException.cpp (hunk 3 above)
Comment (rish9511) [reply 3rdparty/ErrorHandling/MavenException.cpp]: @sahil21 
I plan to implement these levels in the next version
Will add to TODO

File: 3rdparty/ErrorHandling/MavenException.cpp (hunk 1 above)
Comment (sahil21) [reply 3rdparty/ErrorHandling/MavenException.cpp]: @rish9511 Got it.

File: 3rdparty/ErrorHandling/MavenException.cpp (hunk 3 above)
Comment (sahil21) [reply 3rdparty/ErrorHandling/MavenException.cpp]: @rish9511 Thanks

**Pull request merged**